src/cljs/main/broadfcui/page/test_login.cljs

```diff
@@ -0,0 +1,50 @@
+(ns broadfcui.page.test-login
+  (:require
+   [dmohs.react :as react]
+   [broadfcui.components.buttons :as buttons]
+   [broadfcui.components.foundation-tooltip :as tooltip]
+   [broadfcui.common.icons :as icons]
+   [broadfcui.common.input :as input]
+   [broadfcui.common.style :as style]
+   [broadfcui.nav :as nav]
+   [broadfcui.utils :as utils]
+   [broadfcui.utils.user :as user]
+   ))
+
+(react/defc Page
+  {:render
+   (fn [{:keys [this refs]}]
+    [:div {}
+      [:div {}
+        "Token:"
+        [input/TextField {:ref "force-token"
+                          :autoFocus true
+                          :type "password"
+                          :data-test-id "force-token"
+                          :style {:width "85%" :margin "0 1rem"}}]
+        [buttons/Button {:text "Force Sign-In" :onClick
+        (fn [e]
+          (.preventDefault e)
+          (let [[token & fails] (input/get-and-validate refs "force-token")
```

Comment:
Instead of getting the value by ref, why not put it into the state with an `onChange` on the input?

Reply:
no particular reason; this felt like fewer moving parts to me but I don't feel strongly about it at all.